Sources/Commands/PinsStore.swift

```diff
@@ -90,12 +90,14 @@ public struct PinsStore {
 
     /// Unpin a pinnned repository and saves the state.
     ///
+    /// - Precondition: Autopin should be off.
     /// - Parameters:
     ///   - package: The package name to unpin. It should already be pinned.
     /// - Returns: The pin which was removed.
     /// - Throws: PinOperationError
     @discardableResult
     public mutating func unpin(package: String) throws -> Pin {
+        precondition(!autoPin, "Trying to unpin individual package with autopin on.")
```

Comment:
Why not throw here, versus precondition? Client is forced to handle in either case, but throw is a little more obvious.